ReactiveCocoaTests/Swift/SignalProducerSpec.swift

```diff
@@ -72,13 +152,42 @@ class SignalProducerSpec: QuickSpec {
 		}
 
 		describe("SignalProducer.try") {
-			pending("should run the operation once per start()") {
+			it("should run the operation once per start()") {
+				var operationRunTimes = 0
+				let operation: () -> Result<String, NSError> = {
+					operationRunTimes++
+
+					return Result<String, NSError>.Success(Box("OperationValue"))
+				}
+
+				SignalProducer.try(operation).start()
+				SignalProducer.try(operation).start()
+
+				expect(operationRunTimes).to(equal(2))
 			}
 
-			pending("should send the value then complete") {
+			it("should send the value then complete") {
+				let operationReturnValue = "OperationValue"
+				let operation: () -> Result<String, NSError> = {
+					return Result<String, NSError>.Success(Box(operationReturnValue))
+				}
+
+				let signalProducer = SignalProducer.try(operation)
+
+				startSignalProducer(signalProducer, expectSentValue: operationReturnValue, sentError: nil, complete: true)
 			}
 
-			pending("should send the error") {
+			it("should send the error") {
+				let operationError = NSError(domain: "com.reactivecocoa.errordomain", code: 4815, userInfo: nil)
+				let operation: () -> Result<String, NSError> = {
+					return Result<String, NSError>.Failure(Box(operationError))
+				}
+
+				let signalProducer = SignalProducer.try(operation)
+
+				// This is not the current behavior, but should it?
```

Comment:
From [Event.swift](https://github.com/ReactiveCocoa/ReactiveCocoa/blob/8c67470a34dde0b10e296a95a974c48b63990063/ReactiveCocoa/Swift/Event.swift#L17)

> `Next* (Error | Completed)?`

Yep, it's been like this for for a while. I've sometimes considered a unified `Completed` that has an optional error, but I strongly suspect that would be more complicated than it is now.

> if you're just interested in the values, and want to know when it's _done sending values_, you would have to handle both the error and the completion

I've found the existing operators provided good balance so that this has never been a pain point or a source of boilerplate noise, for me personally anyway. I also can't recall many issues/questions about this. The easy solution is to just use: `catchTo(SignalProducer.empty)`.